cloudant-client/src/main/java/com/cloudant/client/api/query/QueryBuilder.java

```diff
@@ -24,36 +24,144 @@
+    private boolean update = true;
+    private boolean stable;
+    private boolean executionStats;
+    private String[] useIndex;
 
+    /**
+     * Construct a {@code QueryBuilder} object for use with {@link com.cloudant.client.api.Database#query(String, Class)}
+     * @param selector JSON object describing criteria used to select documents.
+     * @see <a
+     * href="https://console.bluemix.net/docs/services/Cloudant/api/cloudant_query.html#selector-syntax"
+     * target="_blank">selector syntax</a>
+     * @see com.cloudant.client.api.Database#query(String, Class)
+     */
     public QueryBuilder(Selector selector) {
         this.selector = selector;
     }
 
+    /**
+     * Set the fields option for the query builder.
+     * @param fields List specifying which fields of each object should be returned. If it is
+     *               omitted, the entire object is returned.
+     * @return {@code QueryBuilder} object for method chaining.
+     */
     public QueryBuilder fields(String... fields) {
         this.fields = fields;
         return this;
     }
 
+    /**
+     * Set the sort option for the query builder.
+     * @param sort List specifying sort order of returned documents.
+     * @return {@code QueryBuilder} object for method chaining.
+     */
     public QueryBuilder sort(Sort... sort) {
         this.sort = sort;
         return this;
     }
 
+    /**
+     * Set the limit option for the query builder.
+     * @param limit Maximum number of results returned. Default is 25.
+     * @return {@code QueryBuilder} object for method chaining.
+     */
     public QueryBuilder limit(long limit) {
         this.limit = limit;
         return this;
     }
 
+    /**
+     * Set the skip option for the query builder.
+     * @param skip Skip the first {@code n} results, where {@code n} is the value specified.
+     * @return {@code QueryBuilder} object for method chaining.
+     */
     public QueryBuilder skip(long skip) {
         this.skip = skip;
         return this;
     }
 
+    /**
+     * Set the bookmark option for the query builder.
+     * @param bookmark  A string that enables you to specify which page of results you require.
+     *                  Used for paging through result sets. Every query returns an opaque string
+     *                  under the bookmark key that can then be passed back in a query to get the
+     *                  next page of results. If any part of the selector query changes between
+     *                  requests, the results are undefined.
+     * @return {@code QueryBuilder} object for method chaining.
+     */
+    public QueryBuilder bookmark(String bookmark) {
+        this.bookmark = bookmark;
+        return this;
+    }
+
+    /**
+     * Set the update option for the query builder.
+     * @param update Whether to update the index prior to returning the result. Default is {@code
+     * true}.
+     * @return {@code QueryBuilder} object for method chaining.
```

Comment:
If we can link to Cloudant/Couch documentation for update (and stable) it would be good - I think they are less ambiguous than `stale` was, but I think it would be helpful.

Reply:
Documentation [here](http://docs.couchdb.org/en/2.1.1/api/database/find.html) is minimal, which is the content I pasted in to the `@params`. update/stale/stable/ are not documented at all at https://console.bluemix.net/docs/services/Cloudant/api/cloudant_query.html#query-parameters